src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/TelemetryPolicy.cs

```diff
@@ -11,35 +11,56 @@ namespace Azure.Core.Pipeline.Policies
 {
     public class TelemetryPolicy : HttpPipelinePolicy
     {
-        private readonly HttpHeader _header;
+        private readonly Assembly _clientAssembly;
 
-        public TelemetryPolicy(Assembly clientAssembly, string applicationId)
+        private HttpHeader _header;
+
+        private string _applicationId;
+
+        public string ApplicationId
```

Comment:
As we talked, this really needs to be project wide setting. It does not make sense to change it on per policy instance basis. Having said that, it might be worth to think about a hierarchy, i.e. if a service client is used in a library, which in turn is used in an app, it would be great if the "application ID" preserved this hierarchy.

cc: @JeffreyRichter, @JonathanGiles, @bterlson, @johanste

Reply:
This is true but is outside the scope of this change.

Reply:
please file an issue.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/6189